rust/numaflow-core/src/metrics.rs

```diff
@@ -114,6 +115,7 @@ pub(crate) enum PipelineComponents {
     Source(Source),
     Sink(SinkWriter),
     Map(MapHandle),
+    Reduce(UserDefinedAlignedReduce),
```

Comment:
once we add Unaligned, this will be generic over Aligned/Unaliged Reduce, right or planning to do `enum`?

Reply:
Planning to go with enum for this.